feature/session/src/main/java/io/github/droidkaigi/confsched2019/session/ui/item/SpeakerItem.kt

```diff
@@ -64,19 +65,17 @@ class SpeakerItem @AssistedInject constructor(
         }
     }
 
-    override fun equals(other: Any?): Boolean {
-        if (this === other) return true
-        if (javaClass != other?.javaClass) return false
-
-        other as SpeakerItem
+    override fun providerEqualableContents(): Array<*> = arrayOf(speaker)
 
-        if (speaker != other.speaker) return false
-        if (query != other.query) return false
+    override fun isTextHighlightNeedUpdate() = query?.let {
+        speaker.name.toLowerCase().contains(it.toLowerCase().toRegex())
```

Comment:
Sorry, this `toRegex()` is not good :bowing_man: 
Type `+` crashes app, I will fix it.